test/parallel/test-readline-interface.js

```diff
@@ -115,6 +115,47 @@ function assertCursorRowsAndCols(rli, rows, cols) {
     code: 'ERR_INVALID_ARG_VALUE'
   });
 
+  // Constructor throws if history is not an array
+  assert.throws(() => {
+    readline.createInterface({
+      input,
+      history: 'not an array'
+    });
+  }, {
+    name: 'TypeError',
+    code: 'ERR_INVALID_ARG_VALUE'
+  });
+
+  assert.throws(() => {
+    readline.createInterface({
+      input,
+      history: 123
+    });
+  }, {
+    name: 'TypeError',
+    code: 'ERR_INVALID_ARG_VALUE'
+  });
+
+  assert.throws(() => {
+    readline.createInterface({
+      input,
+      history: {}
+    });
+  }, {
+    name: 'TypeError',
+    code: 'ERR_INVALID_ARG_VALUE'
+  });
```

Comment:
Can you refactor this to avoid code duplication. Something like:

```js
['not an array', 123, 123n, {}, true, Symbol(), null].forEach(history => {
  assert.throws(() => {
    readline.createInterface({
      input,
      history,
    });
  }, {
    name: 'TypeError',
    code: 'ERR_INVALID_ARG_VALUE'
  });

Reply:
Done